build/index.js

```diff
@@ -1,265 +1,42 @@
-        "npm",
-        ["--no-update-notifier", "--no-fund", ...args],
-        { ...process.env, FORCE_COLOR: 1 }
-    );
-    const elapsed = new Date() - start;
-    const realCommand = args[args.length - 1];
-    logDetail(` ${ANSIInvert("npm " + realCommand)} finished in ${elapsed} ms`);
-};
-
-async function npmInstall(name) {
-    if (fs.existsSync("package-lock.json")) {
-        logDetail(`${name}: Installing dependencies ...`);
-        await npm("--no-audit", "--prefer-offline", "ci");
-    }
-}
-
-async function npmRun(script, pkg) {
-    if (!pkg.scripts.includes(script)) return false;
-
-    logDetail(`${pkg.name}: Executing ${script} script ...`);
-    await npm("run", script);
-    return true;
-}
-
-async function cleanPackage(pkg) {
-    if (await npmRun("clean", pkg)) return;
-
-    // Just so that we don't have to add a 'clean' script to every single
-    // package, we cover the conventional case as a fallback - but it's
-    // arguably a bit dangerous.
-    logDetail(`${pkg.name}: Executing fallback clean logic ...`);
-    for (let path of ["node_modules", "dist", "build"])
-        await fsPromises.rm(path, { recursive: true, force: true });
-}
-
-async function syncPackage(pkg) {
-    logDetail(`🕑 ${pkg.name} ...`);
-    if (fs.existsSync("package-lock.json")) {
-        fs.rmSync("package-lock.json");
-    }
-    await npm("i");
-}
-
-const commands = {
-    build: (pkg) => npmInstall(pkg.name).then(() => npmRun("build", pkg)),
-    test: (pkg) => npmRun("test", pkg),
-    clean: (pkg) => cleanPackage(pkg),
-    "sync-deps": (pkg) => syncPackage(pkg),
-};
-
-async function executeCommand(pkg, command) {
-    const oldPath = process.cwd();
-    try {
-        process.chdir(pkg.path);
-    } catch {
-        throw `Directory ${pkg.path} for package ${pkg.name} does not exist`;
-    }
-
-    try {
-        await commands[command](pkg);
-    } finally {
-        process.chdir(oldPath);
-    }
-}
+const { checkVersions, ANSIBold } = require("./utils.js");
+const { logMain, logDependencies, logSuccess } = require("./log.js");
+const { parseCommandLine, showUsage, parseManifest } = require("./cli.js");
+const { createPackageTree, skipPackages } = require("./deps.js");
+const { npm, npx, npmInstall, runCommand } = require("./npm.js");
 
 async function processPackage(name, packageTree, command) {
     if (!(name in packageTree)) throw `Unable to find package ${name}`;
 
     const pkg = packageTree[name];
-    if (pkg.processed) return;
+    if (pkg.isProcessed) return;
 
     for (let dep of pkg.localDependencies)
         await processPackage(dep, packageTree, command);
 
     if (command === "list") {
-        if (!Object.values(packageTree).some((pkg) => pkg.processed))
+        if (!Object.values(packageTree).some((pkg) => pkg.isProcessed))
             logMain("Listing all packages in build order");
         console.log(`${name} (${pkg.path})`);
-        pkg.processed = true;
+        pkg.setProcessed();
```

Comment:
That's a pretty un-JS way of doing it, explicit getters and setters... Using properties is the normal way, if you need logic you can write property getters and setters - not get/set functions.